### PR TITLE
style: fix Button small size align when customize theme

### LINKED ISSUE #24089: .ant-btn-sm get wrong padding when changed 'font-size-base' and 'line-heigth-base'.
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/Renderz/create-react-app-antd](https://github.com/Renderz/create-react-app-antd)

### Steps to reproduce
```
  //add these to config-overrides.js
  "@font-size-base": '12px',
  "@line-height-base": '1.2'
```
```
yarn start
```

### What is expected?
padding: 3.8px 7px;

### What is actually happening?
padding: 0px 7px;

| Environment | Info |
|---|---|
| antd | 4.2.2 |
| React | 16.13.1 |
| System | Mac OS |
| Browser | Chorme |

### PR DESCRIPTION
**Pull request by afc163**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #24089

### 💡 Background and solution

https://github.com/ant-design/ant-design/commit/dda45e479659fe33f650f1f10110cc9f650d5e59#r39119320

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix small Button align issue when  customize theme.  |
| 🇨🇳 Chinese |  修复定制主题时小号 Button 错位的问题。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/commit/dda45e479659fe33f650f1f10110cc9f650d5e59#r39119320 引起的，@AshoneA 看看移除掉有没有坑。

**Comment by AshoneA** [comment]
有点问题，紧凑模式下 https://preview-24097-ant-design.surge.sh/components/button/#components-button-demo-size 单 icon 的情况下样式不对。

这个 bug 我看着像是 less 那 padding 算的有点问题

**Comment by afc163** [comment]
紧凑模式下小号按钮已经很小了，默认的 14px * 1.66666 会超出高度，导致错位。这里调整为紧凑模式下 icon-only 的图标 size 和标准按钮 size 一致（按钮很很小了，这个差距意义不大）。

File: components/style/themes/default.less [hunk 1]
```diff
@@ -62,15 +62,15 @@
 @text-selection-bg: @primary-color;
 @font-variant-base: tabular-nums;
 @font-feature-settings-base: 'tnum';
-@font-size-base: 14px;
+@font-size-base: 12px;
```
Comment (AshoneA) [inline components/style/themes/default.less:65]: 为啥要改默认的 font-size 和 line-height， 影响很大啊

Comment (afc163) [reply components/style/themes/default.less]: 改错了，已修正。

**Review by AshoneA — APPROVED** [review]

**Pull request merged**